File: examples/scala/src/test/scala/ThenStage.scala
import org.scalatest._
import com.tngtech.jgiven._
import com.tngtech.jgiven.annotation._
import Assertions._

class ThenStage extends Stage[ThenStage] {

  @ExpectedScenarioState
  var result = 0
  
  def the_result_should_be(expectedResult:Int) = {
    assert(result == expectedResult)
  }
  
}

File: example-projects/java26-module/src/test/java/com/tngtech/jgiven/exampleprojects/java26/module/ThenStage.java
package com.tngtech.jgiven.exampleprojects.java21.module;

import com.tngtech.jgiven.Stage;
import com.tngtech.jgiven.annotation.ScenarioState;

/**
 * Public stage that consumes the scenario state produced by
 * {@link PackagePrivateStage}.
 *
 * <p>The {@code @ScenarioState} fields here are package-private on purpose:
 * JGiven must {@link java.lang.reflect.Field#setAccessible(boolean) set them
 * accessible} from the JGiven module, which under JPMS requires the
 * {@code opens} directive in {@code module-info.java}.</p>
 */
public class ThenStage extends Stage<ThenStage> {

    @ScenarioState(required = true)
    String result;

    public void the_result_is(String expected) {
        org.junit.jupiter.api.Assertions.assertEquals(expected, result);
    }
}
